templates/hooks/health_check/sdk_ensure_tags.go.tpl

```diff
@@ -0,0 +1,13 @@
+     r := rm.concreteResource(res)
+     if r.ko == nil {
+             // Should never happen... if it does, it's buggy code.
+             panic("resource manager's EnsureTags method received resource with nil CR object")
+     }
+     defaultTags := ackrt.GetDefaultTags(&rm.cfg, r.ko, md)
+     var existingTags []*svcapitypes.Tag
+     existingTags = r.ko.Spec.Tags
+
+     resourceTags := ToACKTags(existingTags)
+     tags := acktags.Merge(resourceTags, defaultTags)
+     r.ko.Spec.Tags = FromACKTags(tags)
+     return nil
```

Comment:
I'm not entirely sure that we need the ensure_tags hook, I have never seen a controller use it :) The thing is that the common runtime already does this for us https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/runtime/reconciler.go#L308-L313 . But maybe a missing something..

Reply:
You are right, fixed.